hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/model/HoodieFlinkRecord.java

```diff
@@ -96,7 +99,7 @@ public HoodieOperation getOperation() {
   @Override
   protected Comparable<?> doGetOrderingValue(Schema recordSchema, Properties props) {
     String orderingField = ConfigUtils.getOrderingField(props);
-    if (isNullOrEmpty(orderingField)) {
+    if (isNullOrEmpty(orderingField) || recordSchema.getField(orderingField) == null) {
```

Comment:
Is this because of the schema evolution?

Reply:
This is for case there is no preCombine key, and `precombine.field` will be set to `no_precombine`